cmd/kueuectl/app/create/create_localqueue.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package create
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/cli-runtime/pkg/printers"
+
+	"sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/client-go/clientset/versioned"
+	"sigs.k8s.io/kueue/client-go/clientset/versioned/scheme"
+	kueuev1beta1 "sigs.k8s.io/kueue/client-go/clientset/versioned/typed/kueue/v1beta1"
+	"sigs.k8s.io/kueue/cmd/kueuectl/app/util"
+)
+
+const (
+	lqLong    = `Create a local queue with the given name in the specified namespace.`
+	lqExample = `  # Create a local queue 
+  kueuectl create localqueue my-local-queue -c my-cluster-queue`
+)
+
+type LocalQueueOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+
+	DryRunStrategy   util.DryRunStrategy
+	Name             string
+	Namespace        string
+	EnforceNamespace bool
+	ClusterQueue     v1beta1.ClusterQueueReference
+
+	UserSpecifiedClusterQueue string
+
+	Client kueuev1beta1.KueueV1beta1Interface
+
+	PrintObj printers.ResourcePrinterFunc
+
+	genericiooptions.IOStreams
+}
+
+func NewLocalQueueOptions(streams genericiooptions.IOStreams) *LocalQueueOptions {
+	return &LocalQueueOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("created").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewLocalQueueCmd(clientGetter genericclioptions.RESTClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewLocalQueueOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:     "localqueue NAME --clusterqueue=cluster-queue-name [--dry-run=server|client|none]",
```

Comment:
```suggestion
		Use:     "localqueue NAME -c CLUSTER_QUEUE_NAME [--dry-run STRATEGY]",
```

Could we align with Cobra recommendations? Also, could we avoid double management of dry run options?

https://pkg.go.dev/github.com/spf13/cobra#Command

Reply:
I think [--dry-run=**server|client|none**] more  understandable than [--dry-run STRATEGY].

```bash
kueue create localqueue NAME --clusterqueue=CLUSTER_QUEUE_NAME [--dry-run=server|client|none]
```

Reply:
That approach brings us fragmentations to manage options. It is potentially bug.
Also, users can confirm the options in the options section (`--help`).

Reply:
Ok. Fixed.